recharges/tasks.py

```diff
@@ -4,11 +4,88 @@
 from celery.task import Task
 from celery.utils.log import get_task_logger
 
-from .models import Account
+from .models import Account, Recharge
 
 logger = get_task_logger(__name__)
 
 
+def get_token():
+    """
+    Returns the last token entry
+    """
+    account = Account.objects.order_by('created_at').last()
+    return account.token
+
+
+def get_recharge(recharge_id):
+    """
+    Returns the recharge object from its id
+    """
+    recharge = Recharge.objects.get(id=recharge_id)
+    return recharge
+
+
+def prep_login_data():
+    """
+    Constructs the dict needed for hotsocket login
+    """
+    login_data = {'username': settings.HOTSOCKET_API_USERNAME,
+                  'password': settings.HOTSOCKET_API_PASSWORD,
+                  'as_json': True}
+    return login_data
+
+
+def request_hotsocket_login():
+    """
+    Hotsocket login via post request
+    """
+    login_data = prep_login_data()
+    login_post = requests.post("%s/login" % settings.HOTSOCKET_API_ENDPOINT,
+                               data=login_data)
+    return login_post.json()
+
+
+def prep_hotsocket_data(recharge_id):
+    """
+    Constructs the dict needed to make a hotsocket airtime request
+    """
+    recharge = get_recharge(recharge_id)
+    hotsocket_data = {
+        'username': settings.HOTSOCKET_API_USERNAME,
+        'password': settings.HOTSOCKET_API_PASSWORD,
+        'as_json': True,
+        'token': get_token(),
+        'recipient_msisdn': recharge.msisdn,
+        'product_code': 'DATA',
+        'network_code': 'VOD',
+        'denomination': recharge.amount,
+        'reference': recharge.id
+    }  # TODO: check if hotsocket can handle length & chars (Trello)
+    return hotsocket_data
+
+
+def request_hotsocket_recharge(recharge_id):
+    """
+    Makes hotsocket airtime request
+    """
+    hotsocket_data = prep_hotsocket_data(recharge_id)
+    recharge_post = requests.post("%s/recharge" %
+                                  settings.HOTSOCKET_API_ENDPOINT,
+                                  data=hotsocket_data)
+    return recharge_post.json()
+
+
+def update_recharge_status_hotsocket_ref(recharge, result):
+    """
+    Set recharge object status to In Process and save the hotsocket reference.
+    """
+    recharge.status = 1
```

Comment:
The status should already be `1` at this point. Then depending on status code returned, moved to next status.